crates/wash-lib/src/cli/start.rs

```diff
@@ -72,9 +71,93 @@ pub struct StartComponentCommand {
     /// If this flag is omitted, the timeout will be adjusted to 5 seconds to account for component download times
     #[clap(long = "skip-wait")]
     pub skip_wait: bool,
+
+    /// Digest to verify artifact against (if OCI URL is provided for <target>)
+    #[clap(short = 'd', long = "digest")]
+    pub digest: Option<String>,
+
+    /// Allow latest artifact tags (if OCI URL is provided for <target>)
+    #[clap(long = "allow-latest")]
+    pub allow_latest: bool,
+
+    /// OCI username, if omitted anonymous authentication will be used
+    #[clap(long = "user", env = "WASH_REG_USER", hide_env_values = true)]
+    pub user: Option<String>,
+
+    /// OCI password, if omitted anonymous authentication will be used
+    #[clap(long = "password", env = "WASH_REG_PASSWORD", hide_env_values = true)]
+    pub password: Option<String>,
+
+    /// Allow insecure (HTTP) registry connections
+    #[clap(long = "insecure")]
+    pub insecure: bool,
+
+    /// skip the local OCI cache and pull the artifact from the registry to inspect
+    #[clap(long = "no-cache")]
+    pub no_cache: bool,
+}
+
+pub async fn handle_start(cmd: StartCommand) -> Result<CommandOutput> {
+    // TODO: absolutize the path if it's a relative file
+    let component_ref = if cmd.component_ref.starts_with('/') {
+        format!("file://{}", &cmd.component_ref) // prefix with file:// if it's an absolute path
+    } else {
+        cmd.component_ref.to_string()
+    };
+
+    let mut buf = Vec::new();
+    if PathBuf::from(component_ref.clone()).as_path().is_dir() {
+        let mut f = File::open(component_ref.clone())?;
+        f.read_to_end(&mut buf)?;
+    } else {
+        let cache_file = (!cmd.no_cache).then(|| cached_oci_file(&cmd.component_ref.clone()));
+        buf = get_oci_artifact(
+            component_ref.clone(),
+            cache_file,
+            OciPullOptions {
+                digest: cmd.digest.clone(),
+                allow_latest: cmd.allow_latest,
+                user: cmd.user.clone(),
```

Comment:
@brooksmtownsend I just realized this on my second review: this could be a little gnarly if you’re using a private registry. For the command to work you’d need to have creds. Should we have a helper flag that indicates what it is and skips the smart check?

Reply:
@thomastaylor312 maybe I don't understand, are you saying that the smart check wouldn't work if a private registry is used?

Reply:
Imagine a normal deployment strategy in an enterprise. You have an internally accessible registry, generally something like ACR or ECR that your deployment target (kubernetes, bare VMs, etc). Those nodes/VMs are configured to be able to access those registries, but a normal developer might not have access to them, or would have to set credentials. If you don't set credentials, this command will fail.

Contrast that to how it works right now. `wash` doesn't attempt to pull anything down, it just sends the reference to the host, which then tries to pull it. Does that make any sense?

Reply:
Yeah that makes sense now. Basically there's a flaw in assuming that the resources a host has access to is the same set that the user or operator does with `wash` (similarly, it could be the opposite relationship where `wash` has access but wasmCloud doesn't but that's a different problem.)

It sounds like a helper flag that allows you to specify to skip the check and to tell `wash` that the resource is a component/provider would be the best solution, users would rarely need to use it but it's there just in case. Thoughts @thomastaylor312 ?

Strawman, something like `--type component|provider`? In the help text we can indicate that this skips the initial pull

Reply:
I thought about that too, but my other concern is with bandwidth/download time. With providers in particular they can be quite large so there is a noticeable delay while it gets downloaded and you're essentially downloading twice (once locally and once on the host)